### ISSUE TITLE
error in lit parser?

### ISSUE THREAD

**Issue by alanz** [ucsd-progsys/liquid-fixpoint#296]
The `litP` parser has a comment saying that it defines bit vector literals

```haskell
-- | BitVector literal: lit "#x00000001" (BitVec (Size32 obj))
litP :: Parser Expr
litP = do reserved "lit"
          l <- stringLiteral
          t <- sortP
          return $ ECon $ L (T.pack l) t
```

But is uses a `sortP` for the value. 

Should it use `bvSortP` instead? At https://github.com/ucsd-progsys/liquid-fixpoint/blob/develop/src/Language/Fixpoint/Parse.hs#L372

**Comment by ranjitjhala** [comment]
Oops don't know how this slipped through the crack. Is it still a valid issue @alanz ?

**Comment by alanz** [comment]
The code is unchanged from before, now at https://github.com/ucsd-progsys/liquid-fixpoint/blob/develop/src/Language/Fixpoint/Parse.hs#L476

If it is supposed to be able to parse the example in the comment, then it is still an issue. It was never completely clear to me what the intention of the parser was. So either the documentation needs to be updated, or the parser.

**Comment by atondwal** [comment]
We don't actually have any BitVector tests, do we?

**Comment by ranjitjhala** [comment]
Pretty sure we do...

**Comment by curiousleo** [comment]
> We don't actually have any BitVector tests, do we?

> Pretty sure we do...

There are `BitVec`s in the tests, which I think shows that `BitVec`s are parsed: https://github.com/ucsd-progsys/liquid-fixpoint/blob/3438be3ddf1823aab84b47b0ad34f03640ea384c/tests/pos/hex.ts.fq#L211-L218

However, the comment here was about the `litP` parser being too lenient, i.e. parsing more than it should:

> The litP parser has a comment saying that it defines bit vector literals

My best guess from a cursory reading is that the comment is simply misleading. The name of the parser is `litP` and it parses _any_ literal. The comment "BitVector literal" seems to be a leftover:

- https://github.com/ucsd-progsys/liquid-fixpoint/commit/a288b6307345ab7cdfe71f29b7e06801dc43d9a8 added `litP`, which at the time _only_ parsed `BitVec`. It also added the comment "BitVector literal".
- https://github.com/ucsd-progsys/liquid-fixpoint/commit/4894f8320ba1c50eb609ced15a1fa8d8340e5932 generalised `litP` to accept all sorts, not just `BitVec`, but left the comment.

If that is all correct, then removing the leftover comment about BitVector should resolve the issue.

**Comment by facundominguez** [comment]
`litP` is only used at one place in `funAppP` which doesn't seem specific to bitvectors. So it would be one more bit of evidence to support removing the outdated comment.

**Comment by curiousleo** [comment]
I think this issue can be closed now that #515 has been merged.
